docs/GUIDE.MD

```diff
@@ -0,0 +1,50 @@
+# How To Write Test Harness Tests
+
+_Refer to the Variable Cloud tests and Variable Local tests on how to write tests for both Cloud and Local bucketing._
+
+![Architecture overview of the test harness](https://user-images.githubusercontent.com/10345366/211103174-93507d88-814f-4177-bbb3-392830e6a280.png)
+
+The tests send commands to the proxy server through http requests, the spec of which you can find at `docs/spec.yaml`. The proxy server then calls the SDK based on the request from the test harness and will respond back with the results to the test harness. The SDK is also configured by the test harness to point all API calls to the mock server using `nock`; the proxy server initializes the SDK with options that change the base URL to the urls passed by the proxy server. This allows us to mock API calls from the SDK.
```

Comment:
maybe add a note about the jest testcontainers setup?
also about the jest environment that is automatically setting up the mock server?